[Transactions] NullPointerException during Java Client bootstrap

**Describe the bug**
I have seen this error, using 2.10.0 Java client, and the Java Client fails to create

```
2022-04-28T16:52:55.1107241Z [javatest.batch] [main] INFO org.apache.pulsar.client.impl.ConnectionPool - [[id: 0x43934007, L:/127.0.0.1:51158 - R:localhost/127.0.0.1:6650]] Connected to server
2022-04-28T16:52:55.1108800Z [javatest.batch] [main] WARN org.apache.pulsar.client.impl.ConnectionHandler - [persistent://pulsar/system/transaction_coordinator_assign-partition-0] [Transaction meta store handler 
[0]] Error connecting to broker: java.lang.NullPointerException
2022-04-28T16:52:55.1112846Z [javatest.batch] [main] ERROR org.apache.pulsar.client.impl.TransactionMetaStoreHandler - Transaction meta handler with transaction coordinator id 0 connection failed.
2022-04-28T16:52:55.1114959Z [javatest.batch] org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.CompletionException: java.lang.NullPointerException
2022-04-28T16:52:55.1117779Z [javatest.batch]   at org.apache.pulsar.client.impl.ConnectionHandler.handleConnectionError(ConnectionHandler.java:89)
2022-04-28T16:52:55.1122767Z [javatest.batch]   at java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:884)
2022-04-28T16:52:55.1124098Z [javatest.batch]   at java.util.concurrent.CompletableFuture.uniExceptionallyStage(CompletableFuture.java:898)
2022-04-28T16:52:55.1124985Z [javatest.batch]   at java.util.concurrent.CompletableFuture.exceptionally(CompletableFuture.java:2209)
2022-04-28T16:52:55.1125838Z [javatest.batch]   at org.apache.pulsar.client.impl.ConnectionHandler.grabCnx(ConnectionHandler.java:74)
2022-04-28T16:52:55.1126556Z [javatest.batch]   at org.apache.pulsar.client.impl.TransactionMetaStoreHandler.<init>(TransactionMetaStoreHandler.java:105)
2022-04-28T16:52:55.1127762Z [javatest.batch]   at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.lambda$startAsync$0(TransactionCoordinatorClientImpl.java:89)
2022-04-28T16:52:55.1128565Z [javatest.batch]   at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
2022-04-28T16:52:55.1129220Z [javatest.batch]   at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
2022-04-28T16:52:55.1130044Z [javatest.batch]   at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.startAsync(TransactionCoordinatorClientImpl.java:78)
2022-04-28T16:52:55.1131008Z [javatest.batch]   at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.start(TransactionCoordinatorClientImpl.java:68)
2022-04-28T16:52:55.1131980Z [javatest.batch]   at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:205)
2022-04-28T16:52:55.1132693Z [javatest.batch]   at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:141)
2022-04-28T16:52:55.1133327Z [javatest.batch]   at org.apache.pulsar.client.impl.ClientBuilderImpl.build(ClientBuilderImpl.java:67)
2022-04-28T16:52:55.1134365Z [javatest.batch]   at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.ensureInitialized(PulsarConnectionFactory.java:360)
2022-04-28T16:52:55.1135204Z [javatest.batch]   at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.createConnection(PulsarConnectionFactory.java:452)
2022-04-28T16:52:55.1136173Z [javatest.batch]   at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.createConnection(PulsarConnectionFactory.java:55)
2022-04-28T16:52:55.1136963Z [javatest.batch]   at com.sun.ts.tests.jms.common.JmsTool.createCommonQSetup(JmsTool.java:754)
2022-04-28T16:52:55.1137933Z [javatest.batch]   at com.sun.ts.tests.jms.common.JmsTool.<init>(JmsTool.java:288)
2022-04-28T16:52:55.1138679Z [javatest.batch]   at com.sun.ts.tests.jms.core20.appclient.messageconsumertests.Client.queueSendRecvMessageListenerTest(Client.java:199)
2022-04-28T16:52:55.1139429Z [javatest.batch]   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-04-28T16:52:55.1140004Z [javatest.batch]   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-04-28T16:52:55.1141163Z [javatest.batch]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-28T16:52:55.1142740Z [javatest.batch]   at java.lang.reflect.Method.invoke(Method.java:498)
2022-04-28T16:52:55.1143147Z [javatest.batch]   at com.sun.ts.lib.harness.EETest.run(EETest.java:596)
2022-04-28T16:52:55.1143574Z [javatest.batch]   at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:115)
2022-04-28T16:52:55.1144079Z [javatest.batch]   at com.sun.ts.tests.common.vehicle.EmptyVehicleRunner.run(EmptyVehicleRunner.java:40)
2022-04-28T16:52:55.1145130Z [javatest.batch]   at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:105)
2022-04-28T16:52:55.1145775Z [javatest.batch]   at com.sun.ts.lib.harness.EETest.getPropsReady(EETest.java:486)
2022-04-28T16:52:55.1146223Z [javatest.batch]   at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:209)
2022-04-28T16:52:55.1146643Z [javatest.batch]   at com.sun.ts.lib.harness.EETest.run(EETest.java:285)
2022-04-28T16:52:55.1147101Z [javatest.batch]   at com.sun.ts.tests.common.vehicle.VehicleClient.main(VehicleClient.java:38)
2022-04-28T16:52:55.1148050Z [javatest.batch] Caused by: java.util.concurrent.CompletionException: java.lang.NullPointerException
2022-04-28T16:52:55.1148744Z [javatest.batch]   at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
2022-04-28T16:52:55.1149345Z [javatest.batch]   at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
2022-04-28T16:52:55.1149919Z [javatest.batch]   at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:673)
2022-04-28T16:52:55.1150480Z [javatest.batch]   at java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:683)
2022-04-28T16:52:55.1151030Z [javatest.batch]   at java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2010)
2022-04-28T16:52:55.1151786Z [javatest.batch]   at org.apache.pulsar.client.impl.ConnectionHandler.grabCnx(ConnectionHandler.java:73)
2022-04-28T16:52:55.1152194Z [javatest.batch]   ... 27 more
2022-04-28T16:52:55.1152482Z [javatest.batch] Caused by: java.lang.NullPointerException
2022-04-28T16:52:55.1153066Z [javatest.batch]   at org.apache.pulsar.client.impl.TransactionMetaStoreHandler.connectionOpened(TransactionMetaStoreHandler.java:121)
2022-04-28T16:52:55.1153770Z [javatest.batch]   at org.apache.pulsar.client.impl.ConnectionHandler.lambda$grabCnx$0(ConnectionHandler.java:73)
2022-04-28T16:52:55.1154499Z [javatest.batch]   at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:670)
2022-04-28T16:52:55.1154874Z [javatest.batch]   ... 30 more
2022-04-28T16:52:55.1156179Z [javatest.batch] [main] WARN org.apache.pulsar.client.impl.ConnectionHandler - [persistent://pulsar/system/transaction_coordinator_assign-partition-0] [Transaction meta store handler [0]] Could not get connection to broker: java.lang.NullPointerException -- Will try again in 0.1 s
2022-04-28T16:52:55.1157139Z [javatest.batch] [main] ERROR org.apache.pulsar.client.impl.PulsarClientImpl - Start transactionCoordinatorClient error.
```

**To Reproduce**
I am not able to reproduce it consistently

**Expected behavior**
No error

new stacktrace, based on 2.10

> 2022-06-06T20:05:07.6976550Z [javatest.batch] 06-06-2022 20:05:07:  ERROR: javax.jms.JMSException: org.apache.pulsar.client.api.PulsarClientException: org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException: org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.CompletionException: java.lang.NullPointerException
> 2022-06-06T20:05:07.6977572Z [javatest.batch] 	at com.datastax.oss.pulsar.jms.Utils.handleException(Utils.java:63)
> 2022-06-06T20:05:07.6978201Z [javatest.batch] 	at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.ensureInitialized(PulsarConnectionFactory.java:395)
> 2022-06-06T20:05:07.6978935Z [javatest.batch] 	at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.createConnection(PulsarConnectionFactory.java:472)
> 2022-06-06T20:05:07.6979736Z [javatest.batch] 	at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.createConnection(PulsarConnectionFactory.java:58)
> 2022-06-06T20:05:07.6980360Z [javatest.batch] 	at com.sun.ts.tests.jms.common.JmsTool.createCommonQSetup(JmsTool.java:754)
> 2022-06-06T20:05:07.6980842Z [javatest.batch] 	at com.sun.ts.tests.jms.common.JmsTool.<init>(JmsTool.java:288)
> 2022-06-06T20:05:07.6981392Z [javatest.batch] 	at com.sun.ts.tests.jms.core20.appclient.jmscontextqueuetests.Client.setup(Client.java:153)
> 2022-06-06T20:05:07.6981952Z [javatest.batch] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 2022-06-06T20:05:07.6982473Z [javatest.batch] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 2022-06-06T20:05:07.6983059Z [javatest.batch] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 2022-06-06T20:05:07.6983555Z [javatest.batch] 	at java.lang.reflect.Method.invoke(Method.java:498)
> 2022-06-06T20:05:07.6983955Z [javatest.batch] 	at com.sun.ts.lib.harness.EETest.run(EETest.java:569)
> 2022-06-06T20:05:07.6984443Z [javatest.batch] 	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:115)
> 2022-06-06T20:05:07.6984974Z [javatest.batch] 	at com.sun.ts.tests.common.vehicle.EmptyVehicleRunner.run(EmptyVehicleRunner.java:40)
> 2022-06-06T20:05:07.6985493Z [javatest.batch] 	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:105)
> 2022-06-06T20:05:07.6985953Z [javatest.batch] 	at com.sun.ts.lib.harness.EETest.getPropsReady(EETest.java:486)
> 2022-06-06T20:05:07.6986400Z [javatest.batch] 	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:209)
> 2022-06-06T20:05:07.6986833Z [javatest.batch] 	at com.sun.ts.lib.harness.EETest.run(EETest.java:285)
> 2022-06-06T20:05:07.6987298Z [javatest.batch] 	at com.sun.ts.tests.common.vehicle.VehicleClient.main(VehicleClient.java:38)
> 2022-06-06T20:05:07.6988304Z [javatest.batch] Caused by: org.apache.pulsar.client.api.PulsarClientException: org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException: org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.CompletionException: java.lang.NullPointerException
> 2022-06-06T20:05:07.6989304Z [javatest.batch] 	at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:212)
> 2022-06-06T20:05:07.6989856Z [javatest.batch] 	at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:145)
> 2022-06-06T20:05:07.6990425Z [javatest.batch] 	at org.apache.pulsar.client.impl.ClientBuilderImpl.build(ClientBuilderImpl.java:67)
> 2022-06-06T20:05:07.6991107Z [javatest.batch] 	at com.datastax.oss.pulsar.jms.PulsarConnectionFactory.ensureInitialized(PulsarConnectionFactory.java:380)
> 2022-06-06T20:05:07.6991572Z [javatest.batch] 	... 17 more
> 2022-06-06T20:05:07.6992324Z [javatest.batch] Caused by: org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException: org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.CompletionException: java.lang.NullPointerException
> 2022-06-06T20:05:07.6993420Z [javatest.batch] 	at org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException.unwrap(TransactionCoordinatorClientException.java:131)
> 2022-06-06T20:05:07.6994384Z [javatest.batch] 	at org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException.unwrap(TransactionCoordinatorClientException.java:129)
> 2022-06-06T20:05:07.6995299Z [javatest.batch] 	at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.start(TransactionCoordinatorClientImpl.java:74)
> 2022-06-06T20:05:07.6996013Z [javatest.batch] 	at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:209)
> 2022-06-06T20:05:07.6996404Z [javatest.batch] 	... 20 more
> 2022-06-06T20:05:07.6996925Z [javatest.batch] Caused by: org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.CompletionException: java.lang.NullPointerException
> 2022-06-06T20:05:07.6997815Z [javatest.batch] 	at org.apache.pulsar.client.impl.ConnectionHandler.handleConnectionError(ConnectionHandler.java:89)
> 2022-06-06T20:05:07.6998472Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:884)
> 2022-06-06T20:05:07.6999098Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.uniExceptionallyStage(CompletableFuture.java:898)
> 2022-06-06T20:05:07.6999711Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.exceptionally(CompletableFuture.java:2209)
> 2022-06-06T20:05:07.7000300Z [javatest.batch] 	at org.apache.pulsar.client.impl.ConnectionHandler.grabCnx(ConnectionHandler.java:74)
> 2022-06-06T20:05:07.7000946Z [javatest.batch] 	at org.apache.pulsar.client.impl.TransactionMetaStoreHandler.<init>(TransactionMetaStoreHandler.java:108)
> 2022-06-06T20:05:07.7001755Z [javatest.batch] 	at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.lambda$startAsync$0(TransactionCoordinatorClientImpl.java:93)
> 2022-06-06T20:05:07.7012540Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
> 2022-06-06T20:05:07.7013354Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
> 2022-06-06T20:05:07.7014134Z [javatest.batch] 	at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.startAsync(TransactionCoordinatorClientImpl.java:82)
> 2022-06-06T20:05:07.7015046Z [javatest.batch] 	at org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.start(TransactionCoordinatorClientImpl.java:72)
> 2022-06-06T20:05:07.7015588Z [javatest.batch] 	... 21 more
> 2022-06-06T20:05:07.7016001Z [javatest.batch] Caused by: java.util.concurrent.CompletionException: java.lang.NullPointerException
> 2022-06-06T20:05:07.7016580Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
> 2022-06-06T20:05:07.7017182Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
> 2022-06-06T20:05:07.7017775Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:673)
> 2022-06-06T20:05:07.7018349Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:683)
> 2022-06-06T20:05:07.7018926Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2010)
> 2022-06-06T20:05:07.7019490Z [javatest.batch] 	at org.apache.pulsar.client.impl.ConnectionHandler.grabCnx(ConnectionHandler.java:73)
> 2022-06-06T20:05:07.7019901Z [javatest.batch] 	... 27 more
> 2022-06-06T20:05:07.7020215Z [javatest.batch] Caused by: java.lang.NullPointerException
> 2022-06-06T20:05:07.7020807Z [javatest.batch] 	at org.apache.pulsar.client.impl.TransactionMetaStoreHandler.connectionOpened(TransactionMetaStoreHandler.java:124)
> 2022-06-06T20:05:07.7021510Z [javatest.batch] 	at org.apache.pulsar.client.impl.ConnectionHandler.lambda$grabCnx$0(ConnectionHandler.java:73)
> 2022-06-06T20:05:07.7022091Z [javatest.batch] 	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:670)
> 2022-06-06T20:05:07.7022467Z [javatest.batch] 	... 30 more

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java
@@ -105,9 +105,12 @@ public class TransactionMetaStoreHandler extends HandlerState
                 .create(),
             this);
         this.connectFuture = connectFuture;
-        this.connectionHandler.grabCnx();
+        this.internalPinnedExecutor = pulsarClient.getInternalExecutorService();
         this.timer = pulsarClient.timer();
-        internalPinnedExecutor = pulsarClient.getInternalExecutorService();
+    }
+
+    public void start() {
+        this.connectionHandler.grabCnx();
     }
 
     @Override

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java
@@ -94,6 +94,7 @@ public class TransactionCoordinatorClientImpl implements TransactionCoordinatorC
                                     i, pulsarClient, getTCAssignTopicName(i), connectFuture);
                             handlers[i] = handler;
                             handlerMap.put(i, handler);
+                            handler.start();
                         }
                     } else {
                         handlers = new TransactionMetaStoreHandler[1];
@@ -103,6 +104,7 @@ public class TransactionCoordinatorClientImpl implements TransactionCoordinatorC
                                 getTCAssignTopicName(-1), connectFuture);
                         handlers[0] = handler;
                         handlerMap.put(0, handler);
+                        handler.start();
                     }
 
                     STATE_UPDATER.set(TransactionCoordinatorClientImpl.this, State.READY);

